pyomo/contrib/pyros/util.py

```diff
@@ -0,0 +1,683 @@
+    :param model: deterministic model
+    :param config: config block
+    :return:
+    '''
+
+    m = model
+    obj = [o for o in model.component_data_objects(Objective)]
+    if len(obj) > 1:
+        raise AttributeError("Deterministic model must only have 1 active objective!")
+    if obj[0].sense != minimize:
+        raise AttributeError("PyROS requires deterministic models to have an objective function with  'sense'=minimization. "
+                             "Please specity your objective function as minimization.")
+    first_stage_terms = []
+    second_stage_terms = []
+
+    first_stage_cost_expr = 0
+    second_stage_cost_expr = 0
+    const_obj_expr = 0
+
+    if isinstance(obj[0].expr, Var):
+        obj_to_parse = [obj[0].expr]
+    else:
+        obj_to_parse = obj[0].expr.args
+    first_stage_variable_set = ComponentSet(model.util.first_stage_variables)
+    second_stage_variable_set = ComponentSet(model.util.second_stage_variables)
+    for term in obj_to_parse:
+        vars_in_term = list(v for v in identify_variables(term))
+
+        first_stage_vars_in_term = list(v for v in vars_in_term if
+                                        v in first_stage_variable_set)
+        second_stage_vars_in_term = list(v for v in vars_in_term if
+                                         v in second_stage_variable_set)
+        for v in first_stage_vars_in_term:
+            if id(v) not in list(id(var) for var in first_stage_terms):
+                first_stage_terms.append(v)
+        for v in second_stage_vars_in_term:
+            if id(v) not in list(id(var) for var in second_stage_terms):
+                second_stage_terms.append(v)
+
+        if len(first_stage_vars_in_term) > 0 and len(second_stage_vars_in_term) > 0:
+            second_stage_cost_expr += term
+        elif len(first_stage_vars_in_term) > 0 and len(second_stage_vars_in_term) == 0:
+            first_stage_cost_expr += term
+        elif len(first_stage_vars_in_term) == 0 and len(second_stage_vars_in_term) > 0:
+            second_stage_cost_expr += term
+        elif len(vars_in_term) == 0:
+            const_obj_expr += term
+    m.first_stage_objective = Expression(expr=first_stage_cost_expr)
+    m.second_stage_objective = Expression(expr=second_stage_cost_expr)
+    m.const_obj_term = Expression(expr=const_obj_expr)
+    return
+
+
+def load_final_solution(model_data, master_soln):
+    '''
+    load the final solution into the original model object
+    :param model_data: model data container object
+    :param master_soln: results data container object returned to user
+    :return:
+    '''
+    model = model_data.original_model
+    soln = master_soln.nominal_block
+    for v in model.component_data_objects(Var, descend_into=True):
+        for soln_v in soln.component_data_objects(Var, descend_into=True):
+            if v.local_name == soln_v.local_name:
+                v.set_value(value(soln_v))
```

Comment:
In a previous comment there is code for building the mapping between two models (generated by clone()).  We can use the same approach here.